src/analysisd/mitre.c

```diff
@@ -0,0 +1,147 @@
+        merror("Unable to connect to socket '%s'", WDB_LOCAL_SOCK);
+        goto end;
+    }
+
+    if (result == -1) {
+        merror("No response or bad response from wazuh-db: '%s'", response);
+        goto end;
+    }
+
+    /* Parse IDs string */
+    const char *jsonErrPtr;
+    if (root = cJSON_ParseWithOpts(response+3, &jsonErrPtr, 0), !root) {
+        merror("Response from the Mitre database cannot be parsed: '%s'", response);
+        result = -1;
+        goto end;
+    }
+
+    /* Getting array size */
+    if (size_ids = cJSON_GetArraySize(root), size_ids == 0) {
+        merror("Response from the Mitre database has 0 elements.");
+        result = -1;
+        goto end;
+    }
+
+    for (i = 0; i < size_ids; i++){
+        /* Getting Mitre attack ID from Mitre's database in Wazuh-DB  */
+        ids = cJSON_GetArrayItem(root, i);
+        if (id = cJSON_GetObjectItem(ids,"id"), id == NULL) {
+            merror("It was not possible to get Mitre techniques information.");
+            result = -1;
+            goto end; 
+        }
+        ext_id = id->valuestring;
+
+        /* Consulting Mitre's database to get Tactics */
+        snprintf(wazuhdb_query, OS_SIZE_6144, "mitre sql SELECT phase_name FROM has_phase WHERE attack_id = '%s';", ext_id);
+        if (result = wdbc_query_ex(&sock, wazuhdb_query, response, OS_SIZE_6144), result == -2) {
+            merror("Unable to connect to socket '%s'", WDB_LOCAL_SOCK);
+            goto end;
+        }
+
+        if (result == -1) {
+            merror("No response or bad response from wazuh-db: '%s'", response);
+            goto end;
+        }
+
+        /* Getting tactics from Mitre's database in Wazuh-DB */
+        tactics_array = cJSON_CreateArray();
+        if (tactics_json = cJSON_ParseWithOpts(response+3, &jsonErrPtr, 0), !tactics_json) {
+            merror("Response from the Mitre database cannot be parsed: '%s'", response);
+            result = -1;
+            goto end;
+        }
+
+        if (size_tactics = cJSON_GetArraySize(tactics_json), size_tactics == 0) {
+            merror("Response from the Mitre database has 0 elements.");
+            result = -1;
+            goto end;
+        }
+        for (j = 0; j < size_tactics; j++) {
+            tactics = cJSON_GetArrayItem(tactics_json, j);
+            if (tactic = cJSON_GetObjectItem(tactics,"phase_name"), tactic == NULL) {
+                merror("It was not possible to get MITRE tactics information.");
+                result = -1;
+                goto end;
+            }
+            cJSON_AddItemToArray(tactics_array, cJSON_Duplicate(tactic,1));
+        }
+        cJSON_Delete(tactics_json);
+
+        /* Filling Hash table with Mitre's information */
+        if (hashcheck = OSHash_Add(mitre_table, ext_id, tactics_array), hashcheck == 0) {
+            merror("Mitre Hash table adding failed. Mitre Technique ID '%s' cannot be stored.", ext_id);
+            result = -1;
+            goto end;
```

Comment:
At this point, going to `end` with `tactics_json ≠ null` non-null and `result = -1`. This will produce a double free (`cJSON_Delete(tactics_json)`).